src/dsp_tools/utils/xmlupload/resource_create_client.py

```diff
@@ -185,88 +199,55 @@ def _make_color_value(value: XMLValue) -> dict[str, Any]:
-    entire_pattern = r"""
-    ^
-    (?:(GREGORIAN|JULIAN|ISLAMIC):)?        # optional calendar
-    (?:(CE|BCE|BC|AD):)?                    # optional era
-    (\d{4}(?:-\d{1,2})?(?:-\d{1,2})?)       # date
-    (?::(CE|BCE|BC|AD))?                    # optional era
-    (?::(\d{4}(?:-\d{1,2})?(?:-\d{1,2})?))? # optional date
-    $
-"""
-    assert isinstance(value.value, str)
-    date_match = regex.search(entire_pattern, value.value, flags=regex.VERBOSE)
-    assert date_match
-    date_groups = date_match.groups()
-    match date_groups:
-        case (str() | None, str() | None, str(), str() | None, str() | None):
-            # start_era cannot be None, and this provides a useful type hint
-            pass
-        case ("ISLAMIC", era, _, end_era, _) if era or end_era:
-            raise UserError(f"ISLAMIC calendar does not support eras: {value.value}")
+def _validate_boolean(s: str) -> bool:
+    match s:
+        case "True" | "true" | "1" | 1 | True:
+            return True
+        case "False" | "false" | "0" | 0 | False:
+            return False
         case _:
-            raise UserError(f"Could not parse date: {value.value}")
-    res: dict[str, Any] = {"@type": "knora-api:DateValue"}
-    calendar, start_era, start_date, end_era, end_date = date_groups
-    if not calendar:
-        calendar = "GREGORIAN"
-    if not end_date:
-        end_date = start_date
-    if calendar != "ISLAMIC":
-        if not start_era:
-            start_era = "CE"
-        if end_date and not end_era:
-            end_era = "CE"
-
-    year, month, day = _parse_date(start_date)
-    res["knora-api:dateValueHasStartYear"] = int(year)
-    if month:
-        res["knora-api:dateValueHasStartMonth"] = int(month)
-    if day:
-        res["knora-api:dateValueHasStartDay"] = int(day)
-    if end_era:
-        res["knora-api:dateValueHasEndEra"] = end_era
-
-    if calendar:
-        res["knora-api:dateValueHasCalendar"] = calendar
-    if start_era:
-        res["knora-api:dateValueHasStartEra"] = start_era
-    if end_date:
-        year, month, day = _parse_date(end_date)
-        res["knora-api:dateValueHasEndYear"] = int(year)
-        if month:
-            res["knora-api:dateValueHasEndMonth"] = int(month)
-        if day:
-            res["knora-api:dateValueHasEndDay"] = int(day)
-    return res
+            raise BaseError(f"Could not parse boolean value: {s}")
 
 
-def _parse_date(date: str) -> tuple[str, str | None, str | None]:
-    date_pattern = r"(\d{4})(?:-(\d{1,2}))?(?:-(\d{1,2}))?"
-    date_match = regex.search(date_pattern, date)
-    assert date_match
-    match date_match.groups():
-        case (year, month, day):
-            assert year
-            return year, month, day
-    raise UserError(f"Could not parse date: {date}")
+def _make_date_value(value: XMLValue) -> dict[str, Any]:
+    string_value = _assert_string(value.value)
+    date = parse_date_string(string_value)
```

Comment:
General concern: Now, we still create the `Date` object during the xmlupload, not before. Do you plan to change that as a follow-up task?

Reply:
correct, fail fast will be a separate task. Eventually, this should happen way earlier in the process.